Sources/Transaction/Fee/BlockchainMetaFetcher.swift

```diff
@@ -36,66 +36,61 @@ final class BlockchainMetaFetcher {
         }
     }
 
-    func verifyBlockVersionOrReset(
-        blockVersion: BlockVersion,
-        completion: @escaping () -> Void
-    ) {
-        inner.accessAsync {
-            if $0.metaCache?.blockVersion != blockVersion {
-                self.resetCache {
-                    completion()
-                }
-            }
-        }
-    }
-
     func cachedBlockVersion() -> BlockVersion? {
         inner.accessWithoutLocking.metaCache?.blockVersion
     }
 
     func feeStrategy(
         for feeLevel: FeeLevel,
+        tokenId: TokenId,
         completion: @escaping (Result<FeeStrategy, ConnectionError>) -> Void
     ) {
         switch feeLevel {
         case .minimum:
-            getOrFetchMinimumFee {
+            getOrFetchMinimumFee(tokenId: tokenId) {
                 completion($0.map { fee in
                     FixedFeeStrategy(fee: fee)
                 })
             }
         }
     }
 
-    func cachedFee() -> UInt64? {
-        inner.accessWithoutLocking.metaCache?.minimumFee
+    func cachedFee(tokenId: TokenId = .MOB) -> UInt64? {
+        inner.accessWithoutLocking.metaCache?.minimumFees[tokenId] ??
+            (tokenId == .MOB ? inner.accessWithoutLocking.metaCache?.minimumFee : nil)
     }
 
-    func getOrFetchMinimumFee(completion: @escaping (Result<UInt64, ConnectionError>) -> Void) {
+    func getOrFetchMinimumFee(
+        tokenId: TokenId,
+        completion: @escaping (Result<UInt64, ConnectionError>) -> Void
+    ) {
         fetchCache {
-            if let minimumFee = $0?.minimumFee {
+            if let minimumFee = $0?.minimumFees[tokenId] ??
+               (tokenId == .MOB ? $0?.minimumFee : nil) {
                 completion(.success(minimumFee))
             } else {
                 self.fetchMeta {
                     completion($0.map { cache in
-                        cache.minimumFee
+                        cache.minimumFees[tokenId] ?? cache.minimumFee
```

Comment:
Shouldn't we throw an error here?  I can see using the minimumFee for .MOB, but using the .MOB minimum fee value for a tokenId that's not found would be incorrect, right?

Reply:
will investigate and see what android does, im told by blockchain that this will never happen tho.

Reply:
changed to `cache.minimumFees[tokenId] ?? McConstants.DEFAULT_MINIMUM_FEE` I think this is better behavior but again this is something that should never happen assuming the token is valid.